consensus/core/src/network/tonic_network.rs

```diff
@@ -115,11 +163,44 @@ impl NetworkClient for TonicClient {
             AUTHORITY_INDEX_METADATA_KEY,
             self.context.own_index.value().to_string().parse().unwrap(),
         );
-        let response = client
+        let mut stream = client
             .fetch_blocks(request)
             .await
-            .map_err(|e| ConsensusError::NetworkError(format!("fetch_blocks failed: {e:?}")))?;
-        Ok(response.into_inner().blocks)
+            .map_err(|e| ConsensusError::NetworkError(format!("fetch_blocks failed: {e:?}")))?
+            .into_inner();
+        let mut blocks = vec![];
+        let mut total_fetched_bytes = 0;
+        loop {
+            match stream.message().await {
+                Ok(Some(response)) => {
+                    for b in &response.blocks {
+                        total_fetched_bytes += b.len();
+                    }
+                    blocks.extend(response.blocks);
+                    if total_fetched_bytes > MAX_TOTAL_FETCHED_BYTES {
```

Comment:
Curious if you think this is something that would be useful to pass in as part of the request so the server knows how much to send back. Then if we do exceed that threshold we know it is indeed byzantine behavior?

Reply:
Definitely, this does seem like a more flexible approach. Adding a TODO to `NetworkClient::fetch_blocks()`.